Update dashboard charts on query change

joshuatf opened the pull request:
Fixes #1188 

Assigns the query from props to the state query when updated.

### Detailed test instructions:

1.  Open the dashboard reports `/wp-admin/admin.php?page=wc-admin`.
2.  Change the query params using the date range picker.
3.  Check that the charts are updated when the date range is changed.

joshuatf committed:
Update dashboard charts on query change

psealock reviewed (commented):
Using the datepicker is working as expected. While we are here, however, I think `query` should be removed from `state` for a few reasons.

All other pages use the url as source of truth, and duplicating `query` in state can cause problems. Already, you can see that the bar/line buttons don't work because they update state, but not the url. The charts update if you manually change the url with `&type=bar`.

I agree with this article and think removing `query` from `state` is a good way to avoid `getDerivedStateFromProps`.

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

joshuatf committed:
Remove state derived props

joshuatf commented:
@psealock You're right.  Removed this and added temporary `chartType` in state until a PR for https://github.com/woocommerce/wc-admin/issues/1184 lands.

Aljullu reviewed (approved):
Much better without `getDerivedStateFromProps`. LGTM. :rocket:
